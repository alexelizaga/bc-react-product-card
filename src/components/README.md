# ProductCard

## Option 1

Example:

```
    <ProductCard
        product={ product }
        className="bg-dark text-white"
    >
        <ProductCard.Image className='custom-image' />
        <ProductCard.Title className='text-bold' />
        <ProductCard.Buttons className='custom-buttons' />
    </ProductCard>
```

## Option 2

```
    <ProductCard
        product={ product2 }
        className='bg-dark text-white'
    >
        <ProductImage className='custom-image' style={{ boxShadow: '10px 10px 10px rgba(0,0,0,0.2)' }} />
        <ProductTitle className='text-bold' />
        <ProductButtons className='custom-buttons' />
    </ProductCard>
```
