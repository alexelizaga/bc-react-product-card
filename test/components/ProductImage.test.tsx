import React from 'react';
import renderer from 'react-test-renderer';

import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductImage', () => {
  test('should render with image', () => {
    const wrapper = renderer.create(
        <ProductImage img='https://test-image.png' className='test-class' style={{ borderRadius: '5px' }} />
    );

    expect( wrapper.toJSON() ).toMatchSnapshot();
  });

  test('should show image of product', () => {
    const wrapper = renderer.create(
        <ProductCard product={product2}>
            {
                () => <ProductImage />
            }
        </ProductCard>
    );

    expect( wrapper.toJSON() ).toMatchSnapshot();
  });

});