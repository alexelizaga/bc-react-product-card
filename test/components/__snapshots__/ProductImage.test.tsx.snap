// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage should render with image 1`] = `
<img
  alt="Product"
  className="productImg test-class"
  src="https://test-image.png"
  style={
    Object {
      "borderRadius": "5px",
    }
  }
/>
`;

exports[`ProductImage should show image of product 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="/assets/coffee-mug.png"
  />
</div>
`;
